randomMessageGenerator/script.js

```diff
@@ -0,0 +1,36 @@
+const nouns = ["pasta", "movie", "music", "species", "cake", "color", "island", "salt", "umbrella", "bathtub", "meeting"];
+const adjectives = ["new", "expensive", "rich", "pink", "futuristic", "free", "important", "ugly", "spicey"];
+const verbs = ["create","invent","discover","imagine","notice","complain about","wonder about", "think about"];
+
+function getRandom(wordCollection) {
+    if (!wordCollection.length) return "";
+    else return wordCollection[Math.floor(Math.random() * wordCollection.length)];
+}
+
+function vowelFirstLetter(word) {
+    if (["o","u","e","a","i"].includes(word[0])) return "an";
+    else return "a";
+}
+
+let verb = getRandom(verbs);
+let adjective = getRandom(adjectives);
+let article = vowelFirstLetter(adjective);
+let noun = getRandom(nouns);
+
+const output = (verb, adjective, noun) => {
+if (verb === "" || adjective === "" || noun === "") {console.log("You will have a great day today!")}
+else 
+console.log(`You will ${verb} ${article} ${adjective} ${noun} today.`);
+}
```

Comment:
Oh dope, they're teaching arrow function syntax! That's where the real beauty is.

1) The `let`s should be const
2) `vowelFirstLetter` should be consistent with a function (and more precise) like this and be `getIndefiniteArticle` (I know english isn't your native language, just chiming in
3) empty strings are falsy values in JS
4) You should look into installing prettier & eslint into all of your projects. And then turn on the auto format on save setting in vscode, since it will auto-indent all your stuff. I HIGHLY recommend reading about this now, so early in your journey, as it will make your life so, so much easier as you learn.

Net result of this and ternaries, etc:
```ts
const output = (verb, adjective, noun) => {
  console.log(
    (!verb || !adjective || !noun)
      ? "You will have a great day today!"
      : `You will ${verb} ${article} ${adjective} ${noun} today.`
  );
}
```

But I look at this, and I'm like, this function is really doing 2 things:
- Assembling the output
- Printing it

Let's split that up:
```ts
const getOutput = (verb, adjective, noun) => {
    (!verb || !adjective || !noun)
      ? "You will have a great day today!"
      : `You will ${verb} ${article} ${adjective} ${noun} today.`
  );

console.log(getOutput(verb, adjective, noun));
```

And now the code feels more cleanly split apart =)

Reply:
ahyeah, I get the point that functions should be named after WHAT THEY DO. Makes sense.